plugins/org.eclipse.fordiac.ide.model/src/org/eclipse/fordiac/ide/model/dataimport/SystemImporter.java

```diff
@@ -459,4 +471,32 @@ private void parseApplication(final AutomationSystem automationSystem)
 		});
 	}
 
+	private void finalizeMappingModel() {
+		generateMappedFBConnections();
+	}
+
+	/**
+	 * For each of the mapped FBs that were generated during the mapping parsing
+	 * check if connections need to be generated in resource model.
+	 *
+	 */
+	private void generateMappedFBConnections() {
+		for (final FBNetworkElement fbnEl : mappedFBs) {
+			final FBNetworkElement srcResFb = fbnEl.getOpposite();
+			final Resource res = fbnEl.getResource();
+			fbnEl.getInterface().getOutputs().flatMap(ie -> ie.getOutputConnections().stream()) //
+					.filter(con -> con.getDestinationElement().getResource() == res) //
+					.forEach(con -> res.getFBNetwork().addConnection(createResourceCon(srcResFb, con)));
```

Comment:
I assume you are creating only outgoing connections to avoid duplicates? What if the source of a connection did exist in the resource network but the destination (and mutual connections) did not? What about the output side of InOut variables?

Reply:
resource only connections will be handled specially. For that I'm currently reviewing and reworking the connection parsing code.

Output side of InOut variables is included in the stream provided by getOutputs.